File: evidence/generated-systems/angular/inventory-on-angular/src/app/models/StorageLocation.js

// Define collection and schema for StorageLocation
export interface StorageLocation {
    code:
	type : string
    temperatureControlled:
	type : boolean
    capacity:
	type : String
    capacityUnit:
	type : string
    Warehouse:
	type : Schema.Types.ObjectId
    ParentLocation:
	type : Schema.Types.ObjectId
    ChildLocations:
 	type : [{ type: Schema.Types.ObjectId, ref: 'StorageLocation' }]
    InventoryItems:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InventoryItem' }]
    LocationType:
 	type : String
#
    collection: 'storageLocations'
}

File: evidence/generated-systems/golang/inventory-on-golang/inventory-on-golang/internal/model/StorageLocation.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// StorageLocation Declaration
//==============================================================
type StorageLocation struct {
    gorm.Model
     Code                                    string
    TemperatureControlled                                    bool
    Capacity                                                            string
    CapacityUnit                                    string
    WarehouseId         *uint
    Warehouse           *Warehouse `gorm:"foreignKey:WarehouseId"`
    ParentLocationId         *uint
    ParentLocation           *StorageLocation `gorm:"foreignKey:ParentLocationId"`
     ChildLocations           []StorageLocation `gorm:"foreignKey:ChildLocationsFromStorageLocationId"`
     InventoryItems           []InventoryItem `gorm:"foreignKey:InventoryItemsFromStorageLocationId"`
    LocationType                      LocationType

// parent associations as their child

}

